torchtext/csrc/sentencepiece.cpp

```diff
@@ -39,7 +39,59 @@ struct SentencePiece : torch::CustomClassHolder {
     return processor_.EncodeAsPieces(input);
   }
 
+  std::vector<std::vector<std::string>>
+  NBestEncodeAsPieces(const std::string &input,
+                      const int64_t nbest_size) const {
+    return processor_.NBestEncodeAsPieces(input, nbest_size);
+  }
+
+  std::vector<std::vector<int64_t>>
+  NBestEncodeAsIds(const std::string &input, const int64_t nbest_size) const {
+    const auto vals = processor_.NBestEncodeAsIds(input, nbest_size);
+    std::vector<std::vector<std::int64_t>> ret;
+    for (const auto &vec : vals) {
+      ret.emplace_back(std::vector<std::int64_t>(vec.begin(), vec.end()));
+    }
+    return ret;
+  }
+
+  std::vector<std::string> SampleEncodeAsPieces(const std::string &input,
```

Comment:
With `SampleEncodeAsPieces` and `SampleEncodeAsIds`, we can add them to sentencepiece transforms [here](https://github.com/zhangguanheng66/text/blob/c30188b7b68a50d18bc2908442bd9786b012108a/torchtext/experimental/transforms/transforms.py#L49-L59)